File: app/src/main/java/org/worshipsongs/preference/DatabaseSettingsPreference.java
package org.worshipsongs.preference;

import android.content.Context;
import android.content.Intent;
import android.preference.Preference;
import android.util.AttributeSet;

import org.worshipsongs.activity.DatabaseSettingActivity;

/**
 * Author : Madasamy
 * Version : 3.x.
 */

public class DatabaseSettingsPreference extends Preference
{

    public DatabaseSettingsPreference(Context context, AttributeSet attrs)
    {
        super(context, attrs);
    }

    @Override
    protected void onClick()
    {
        super.onClick();
        getContext().startActivity(new Intent(getContext(), DatabaseSettingActivity.class));
    }

}

File: app/src/main/java/org/worshipsongs/preference/DatabaseSettingsPreference.kt
package org.worshipsongs.preference

import android.content.Context
import android.content.Intent
import android.util.AttributeSet
import androidx.preference.Preference

import org.worshipsongs.activity.DatabaseSettingActivity

/**
 * @author : Madasamy
 * @version : 3.x.
 */

class DatabaseSettingsPreference(context: Context, attrs: AttributeSet) : Preference(context, attrs)
{

    override fun onClick()
    {
        super.onClick()
        context.startActivity(Intent(context, DatabaseSettingActivity::class.java))
    }

}
